spv/verifypayment_test.go

```diff
@@ -203,6 +238,82 @@ func TestSPVEnvelope_VerifyPayment(t *testing.T) {
 			} else {
 				assert.Nil(t, tx)
 			}
+
+			binary, err := testData.Envelope.Bytes()
+			assert.NoError(t, err, "expected no error when creating binary from json")
+
+			mpv, err := spv.NewMerkleProofVerifier(mch)
+			assert.NoError(t, err, "expected no error when creating binary from json")
+
+			opts := append(test.setupOpts, test.overrideOpts...)
+			valid, err := spv.VerifyAncestryBinary(binary, mpv, opts...)
+			if test.expErrBinary != nil {
+				assert.Error(t, err)
+				assert.EqualError(t, errors.Cause(err), test.expErrBinary.Error())
+				assert.False(t, valid)
+			} else {
+				assert.NoError(t, err)
+				assert.True(t, valid)
+			}
+		})
+	}
+
+}
+func TestVerifyAncestryBinary(t *testing.T) {
```

Comment:
this file is a little confusing with the old and new stuff in there, not sure what the best way would be to separate them or make things clearer, thoughts @sirdeggen @Tigh-Gherr ?

Reply:
This test should be untouched in terms of what it's testing, a new function with indentical cases should be made that `assert.NoError`'s its way to the binary verificaiton.